libraries/AC_AttitudeControl/AC_WeatherVane.cpp

```diff
@@ -107,7 +107,7 @@ AC_WeatherVane::AC_WeatherVane(void)
 bool AC_WeatherVane::get_yaw_out(float &yaw_output, const int16_t pilot_yaw, const float hgt, const float roll_cdeg, const float pitch_cdeg, const bool is_takeoff, const bool is_landing)
 {
     Direction dir = (Direction)_direction.get();
-    if ((dir == Direction::OFF) || !allowed || (pilot_yaw != 0) || !is_positive(_gain)) {
+    if ((dir == Direction::OFF) || !allowed || (pilot_yaw != 0) || !is_positive(_gain) || (dir == Direction::TAKEOFF_OR_LAND_ONLY && !(is_takeoff || is_landing))) {
```

Comment:
You don't need this change. Should be good to return via the switch lower down.

Reply:
Infact, better to add to the if on 125.
```
    if ((dir == Direction::OFF) || (dir == Direction::TAKEOFF_OR_LAND_ONLY)) {
        // Disabled in this flight phase
        reset();
        return false;
    }
```

Reply:
I has it there oriinally...but having a comment saying this both enables or disables during takeoff and landing seemed strange...but I moved it for you...and fixed the param docs...thanks for that catch